airflow/configuration.py

```diff
@@ -1487,47 +1500,118 @@ def set(*args, **kwargs) -> None:
     conf.set(*args, **kwargs)
 
 
-def ensure_secrets_loaded() -> List[BaseSecretsBackend]:
-    """
-    Ensure that all secrets backends are loaded.
-    If the secrets_backend_list contains only 2 default backends, reload it.
-    """
-    # Check if the secrets_backend_list contains only 2 default backends
-    if len(secrets_backend_list) == 2:
-        return initialize_secrets_backends()
-    return secrets_backend_list
+class DefaultSecretsBackend(UserList):
+    """List Container which use for store default secrets backends."""
+
+
+@dataclass(frozen=True)
+class SecretsBackendConfig:
```

Comment:
if we provide only one way of configuring, and we "upgrade" legacy config for backward compatibility, maybe we don't need this either?

Reply:
This helper class for

- read single config of backend
   - separate classmethod for current configuration
   - separate classmethod for other cases
- compare backends config
- initialise backend class with config